huxley/api/tests/test_user.py

```diff
@@ -416,3 +416,67 @@ def test_get(self):
             u'times_attended': school.times_attended,
             u'international': school.international,
         })
+
+
+class DelegateUserCreateTestCase(tests.PartialUpdateAPITestCase):
+    url_name = 'api:delegate_detail'
+
+    def setUp(self):
+        self.advisor = models.new_user(username='advisor', password='advisor')
+        self.school = models.new_school(user=self.advisor)
+        self.registration = models.new_registration(school=self.school)
+        self.committee = models.new_committee()
+        self.country = models.new_country()
+        self.assignment = models.new_assignment(
+            registration=self.registration, country=self.country, committee=self.committee)
+        self.delegate = models.new_delegate(
+            school=self.school, assignment=self.assignment)
+        self.delegate.assignment = None
+        self.delegate.save()
+        self.superuser = models.new_user(is_superuser=True)
+        self.params = {'email': 'test@huxley.org'}
+        self.assign_params = {'assignment': self.assignment.id}
+        self.unassign_params = {'assignment': None}
+
+    def test_delegate_no_user(self):
+        self.client.login(username='advisor', password='advisor')
+        response = self.get_response(self.delegate.id, params=self.params)
+        self.assertFalse(User.objects.filter(delegate__id=self.delegate.id).exists())
+
+    def test_delegate_user_create(self):
+        self.client.login(username='advisor', password='advisor')
+        response = self.get_response(self.delegate.id, params=self.assign_params)
+        self.assertTrue(User.objects.filter(delegate__id=self.delegate.id).exists())
+
+    def test_delegate_user_unassign(self):
+        self.client.login(username='advisor', password='advisor')
+        response1 = self.get_response(self.delegate.id, params=self.assign_params)
+        response2 = self.get_response(self.delegate.id, params=self.unassign_params)
+        self.assertTrue(User.objects.filter(delegate__id=self.delegate.id).exists())
+
+
+class DelegateUserDestroyTestCase(tests.DestroyAPITestCase):
+    url_name = 'api:delegate_detail'
+
+    def setUp(self):
+        self.advisor = models.new_user(username='advisor', password='advisor')
+        self.school = models.new_school(user=self.advisor)
+        self.registration = models.new_registration(school=self.school)
+        self.committee = models.new_committee()
+        self.country = models.new_country()
```

Comment:
I don't think you need most of these variables. The test models module usually creates whatever it needs by default, so if you're not gonna be using them in the actual test, no need to manually create.